src/couch_lru.erl

```diff
@@ -16,48 +16,57 @@
 -include_lib("couch/include/couch_db.hrl").
 
 new() ->
-    {gb_trees:empty(), dict:new()}.
-
-insert(DbName, {Tree0, Dict0}) ->
-    Lru = erlang:now(),
-    {gb_trees:insert(Lru, DbName, Tree0), dict:store(DbName, Lru, Dict0)}.
-
-update(DbName, {Tree0, Dict0}) ->
-    case dict:find(DbName, Dict0) of
-    {ok, Old} ->
-        New = erlang:now(),
-        Tree = gb_trees:insert(New, DbName, gb_trees:delete(Old, Tree0)),
-        Dict = dict:store(DbName, New, Dict0),
-        {Tree, Dict};
-    error ->
-        % We closed this database before processing the update.  Ignore
-        {Tree0, Dict0}
+    Updates = ets:new(couch_lru_updates, [ordered_set]),
+    Dbs = ets:new(couch_lru_dbs, [set]),
+    {0, Updates, Dbs}.
+
+insert(DbName, {Count, Updates, Dbs}) ->
+    update(DbName, {Count, Updates, Dbs}).
+
+update(DbName, {Count, Updates, Dbs}) ->
```

Comment:
Though thinking about it would change the logic. Notice now `Count` is a globally incrementing counter. This change would make so that the count is increment per db shard.